testplan/runnable/interactive/http.py

```diff
@@ -46,6 +54,89 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _extract_cases_from_parametrization(param_entry: Dict) -> List[str]:
+    """
+    Given a parametrization entry, extracts the testcases.
+
+    :param param_entry: parametrization entry
+    :return: list of testcase names
+    """
+    cases = []
+    for entry in param_entry["entries"]:
+        if entry["category"] == ReportCategories.TESTCASE:
+            cases.append(entry["name"])
+    return cases
+
+
+def _extract_cases_from_suite(suite_entry: Dict) -> List[str]:
+    """
+    Given a testsuite entry, extracts the testcases.
+
+    :param suite_entry: testsuite entry
+    :return: list of testcase names
+    """
+    cases = []
+    for entry in suite_entry["entries"]:
+        if entry["category"] == ReportCategories.TESTCASE:
+            cases.append(entry["name"])
+        elif entry["category"] == ReportCategories.PARAMETRIZATION:
+            cases.extend(_extract_cases_from_parametrization(entry))
+    return cases
+
+
+def _extract_suites_cases(shallow_entry: Dict) -> Dict[str, List[str]]:
```

Comment:
This function is used to extract a testsuite to testcase list dictionary that is in line with how the suites and cases are prepared for running in the iterative runner of the MultiTest.